cmd/govim/buffer_events.go

```diff
@@ -17,6 +18,19 @@ import (
 
 func (v *vimstate) bufReadPost(args ...json.RawMessage) error {
 	nb := v.currentBufferInfo(args[0])
+	rel, err := filepath.Rel(v.workingDirectory, nb.URI().Filename())
+	if err != nil {
+		v.Logf("bufReadPost: failed to call filepath.Rel(%q, %q): %v", v.workingDirectory, nb.URI().Filename(), err)
+	}
+
+	// If we load a buffer that already had diagnostics reported by gopls, the buffer number must be
+	// updated to ensure that sign placement etc. works.
+	for i, d := range v.diagnosticsCache {
```

Comment:
On reflection, possibly slightly more effective (both in terms of performance but more sort readability) for `diagnosticsCache` to be `map[string]*types.Diagnostic`?

The key can then be the absolute filepath.

Reply:
Yes (I guess you mean `map[string][]*types.Diagnostic`). I'd still need to sort them in a slice for the quick fix list, but I wouldn't need to iterate over all diagnostics when placing signs & highlighting diagnostics.

Reply:
(I recall that a nice side effect of sorting all diagnostics before placing signs as well was that I automatically got a consistent golden file in the tests.)

Reply:
Just had another quick look at this and indeed, both the sign placement and quickfix list needs to be sorted. Or rather "needs" to be sorted as signs only needs to be sorted for them to be testable with golden files, since they rely on the sign id's (placement order) to be consistent.

So from the three options:
- Change to map & convert to sorted slice sort twice (quickfix & signs)
- Change to map & convert once & rewrite sign-related tests so they don't rely on sign id:s
- Leave as is

I lean towards leaving it as is for now. That change could be a separate PR for the future. Thoughts?

Reply:
Yep, agreed.

Reply:
Leave as is for now